text/0000-symbol-name-mangling-v2.md

```diff
@@ -330,48 +543,56 @@ std::iter::Chain<
   std::iter::Once<(u32, usize)>>
 ```
 
-It would make for a long symbol name if this types is used (maybe repeatedly) as a generic argument somewhere. C++ has the same problem with its templates; which is why the Itanium mangling introduces the concept of compression. If a component of a definition occurs more than once, it will not be repeated and instead be emitted as a substitution marker that allows to reconstruct which component it refers to. The scheme proposed here will use the same approach.
+It would make for a long symbol name if this type is used (maybe
+repeatedly) as a generic argument somewhere. C++ has the same problem
+with its templates; which is why the Itanium mangling introduces the
+concept of compression. If a component of a definition occurs more than
+once, it will not be repeated and instead be emitted as a substitution
+marker that allows to reconstruct which component it refers to. The
+scheme proposed here will use the same approach (but with a simpler
+definition).
 
-The exact scheme will be described in detail in the reference level explanation below but it roughly works as follows: As a mangled symbol name is being built or parsed, we build up a dictionary of "substitutions", that is we keep track of things a subsequent occurrence of which could be replaced by a substitution marker. The substitution marker is then the lookup key into this dictionary. The things that are eligible for substitution are (1) all prefixes of absolute paths (including the entire path itself) and (2) all types except for basic types. If a substitutable item is already present in the dictionary it does not generate a new key. Here's an example in order to illustrate the concept:
+The exact scheme will be described in detail in the reference level
+explanation below but it roughly works as follows: As a mangled symbol
+name is being built, we remember the position of every substitutable item
+in the output string, that is, we keep track of things a subsequent
+occurrence of which could be replaced by a back reference.
 
-```
+The things that are eligible for substitution are (1) all prefixes of
+paths (including the entire path itself), (2) all types except for
+basic types, and (3) instances of const data.
```

Comment:
s/instances of const data/type-level constants (array lengths and values passed to const generic params)

cc @varkor for the exact naming here